dora-rs/dora
# Adding support for `Arrow List(*)`  for Python

This is an intermediate Pull Request in supporting more types. that makes it possible to send all basic types (i16, f32, u64...) when using python API.

The main idea is to refactor the `Metadata` struct to allow it to pass a `data_type` field specifying the type. This type is then used when constructing the arrow array.

This work for dora Python Node API and dora Python Operator API. 

I have also added a small optimization that makes removes a round trip from the Python Operator to the Runtime if the vector is smaller than shared memory. 

### To test

You can check the newly updated `python-operator-example` where `np.uint8` downcasting has been removed and the type is passed through the dora event message.
